[CLOUDCRAFT 1192] Update EKS documentation to reflect modern expectations (#28763)
DataDog/documentation: content/en/cloudcraft/getting-started/connect-amazon-eks-cluster-with-cloudcraft.md

By scanning your Amazon EKS clusters, Cloudcraft allows you to generate system architecture diagrams to help visualize your deployed workloads and pods.

> Cloudcraft uses [access entries][1] to grant [Cloudcraft's existing read-only IAM entity role][2] access to the Kubernetes API. That means Cloudcraft requires no special software or agent to be installed on your cluster.

Reviewer: ```suggestion
Cloudcraft uses [access entries][1] to grant [Cloudcraft's existing read-only IAM entity role][2] access to the Kubernetes API. Cloudcraft does not require any special software or agent to be installed on your cluster.
```
